library/alloc/src/ffi/c_str.rs

```diff
@@ -690,6 +693,53 @@ impl CString {
             }),
         }
     }
+
+    /// Attempts to convert a <code>[Vec]<[u8]></code> to a [`CString`].
+    ///
+    /// The input [`Vec`] must contain at least one nul byte.
+    ///
+    /// If the nul byte is not at the end of the input `Vec`, then the `Vec`
+    /// will be truncated so that there is only one nul byte, and that byte
+    /// is at the end.
```

Comment:
It might be good to mention that this doesn't reduce the capacity of the Vec/CString. (Like the [Vec::truncate docs](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.truncate).)

Reply:
Oh wait, never mind: CString stores a Box<[u8]>, not a Vec<u8>, so it does shrink the capacity.

Reply:
That does raise the question of how useful this method is, though, since it most likely ends up reallocating anyway.

Reply:
I suppose the same thing could be accomplished with `Cstr::from_bytes_with_nul()` following by `.into_c_string()`. But being able to do the same thing from a `CString` context would be a kindness. Particularly if the user doesn't think to check the docs for both places when looking for this functionality.